### PR TITLE
Autoplot method

### PR DESCRIPTION
**Pull request by aarathybabu97**
autoplot method mentioned in #89

### CODE REVIEW

**Comment by aarathybabu97** [comment]
The code written in https://github.com/njtierney/conmat/blob/autoplot-method/R/autoplot.R does not seem to work when it is part of the package 
```r
library(conmat)
library(ggplot2)

  polymod_contact_data <- get_polymod_setting_data()
 polymod_survey_data <- get_polymod_population()
 
 setting_models <- fit_setting_contacts(contact_data_list = polymod_contact_data,
                        population = polymod_survey_data)
 
fairfield_age_pop <- abs_age_lga("Fairfield (C)")
 
 fairfield_hh_size <-get_per_capita_household_size(lga = "Fairfield (C)")

 synthetic_settings_5y_fairfield_hh <- predict_setting_contacts(
   population = fairfield_age_pop,
   contact_model = setting_models,
      age_breaks = c(seq(0, 85, by = 5), Inf),
  per_capita_household_size = fairfield_hh_size
 )

 # Plotting synthetic contact matrices across all settings
 
  autoplot(object =  synthetic_settings_5y_fairfield_hh,
           title="Setting specific synthetic contact matrices")
#> Error in `autoplot()`:
#> ! Objects of type predicted_setting_contacts/list not supported by autoplot.
  
 # Work setting specific synthetic contact matrices
  autoplot(object =  synthetic_settings_5y_fairfield_hh$work,
           title="Work")
#> Error in `autoplot()`:
#> ! Objects of type matrix/array/predicted_setting_contacts not supported by autoplot.
Created on 2022-09-19 by the [reprex package](https://reprex.tidyverse.org/) (v2.0.1)
````

**Comment by aarathybabu97** [comment]
However it seems to work when run like a normal function 

```r
library(conmat)
library(ggplot2)

  polymod_contact_data <- get_polymod_setting_data()
 polymod_survey_data <- get_polymod_population()
 
 setting_models <- fit_setting_contacts(contact_data_list = polymod_contact_data,
                        population = polymod_survey_data)
 
fairfield_age_pop <- abs_age_lga("Fairfield (C)")
 
 fairfield_hh_size <-get_per_capita_household_size(lga = "Fairfield (C)")

 synthetic_settings_5y_fairfield_hh <- predict_setting_contacts(
   population = fairfield_age_pop,
   contact_model = setting_models,
      age_breaks = c(seq(0, 85, by = 5), Inf),
  per_capita_household_size = fairfield_hh_size
 )



autoplot.predicted_setting_contacts <- function(object, ...,title="Contact Matrices") {
  if (any(is.element(class(object), "matrix")))
  {
    object %>%
      matrix_to_predictions() %>%
      ggplot2::ggplot(ggplot2::aes(x = age_group_from,
                                   y = age_group_to,
                                   fill = contacts)) +
      ggplot2::geom_tile() +
      ggplot2::coord_fixed() +
      ggplot2::scale_fill_distiller(direction = 1,
                                    trans = "sqrt") +
      ggplot2::theme_minimal() +
      ggplot2::theme(axis.text = ggplot2::element_text(
        size = 6,
        angle = 45,
        hjust = 1
      ))+
      ggplot2::ggtitle(title)
  }
  else
  {
    do.call(patchwork::wrap_plots, lapply(names(object)[names(object) != "all"],
                                          function (x)
                                          {
                                            object[[x]] %>%
                                              matrix_to_predictions() %>%
                                              ggplot2::ggplot(ggplot2::aes(x = age_group_from,
                                                                           y = age_group_to,
                                                                           fill = contacts)) +
                                              ggplot2::geom_tile() +
                                              ggplot2::coord_fixed() +
                                              ggplot2::scale_fill_distiller(direction = 1,
                                                                            trans = "sqrt") +
                                              ggplot2::theme_minimal() +
                                              ggplot2::theme(axis.text = ggplot2::element_text(
                                                size = 6,
                                                angle = 45,
                                                hjust = 1
                                              )) +
                                              ggplot2::ggtitle(x)
                                          })) -> plot
    
    plot +
      patchwork::plot_annotation(
        title = title
      )
  }
}

```
```r
# Plotting synthetic contact matrices across all settings

autoplot(object =  synthetic_settings_5y_fairfield_hh,
         title="Setting specific synthetic contact matrices")

```
Rplot_all


```r

# Work setting specific synthetic contact matrices
autoplot(object =  synthetic_settings_5y_fairfield_hh$work,
         title="Work")

```

Rplot_work

**Comment by njtierney** [comment]
OK awesome!

I've made a couple of small tweaks to the code to help simplify a couple of things.

The main thing I think is that `autoplot` has to be re-exported from `ggplot2`.

Another potential catch was that the code used the argument `matrix` as the first argument, but `autoplot` requires `object` - these names also didn't match the roxygen.

Here's a summary of the changes I made:

- reexport autoplot from ggplot2
- Add a constructor function to make it a little easier to add the class
- make two classes, "conmat_prediction_matrix" for a single matrix, and "conmat_setting_prediction_matrix" for the list of matrices. This means that we can write two `autoplot` functions, one for each of these classes, rather than handling that within `autoplot` - let the object oriented method dispatch handle the complexity for us
- call `plot_matrix` and `plot_setting_matrix` internally in autoplot - to reduce duplicated code and `do.call`

We can see the result below:

``` r
library(conmat)

fairfield <- abs_age_lga("Fairfield (C)")

fairfield_extrapolated <- extrapolate_polymod(
  population = fairfield
)

autoplot(fairfield_extrapolated$home)
```



``` r
autoplot(fairfield_extrapolated)
```



<sup>Created on 2022-09-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.0 (2022-04-22)
#>  os       macOS Monterey 12.3.1
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_AU.UTF-8
#>  ctype    en_AU.UTF-8
#>  tz       Pacific/Auckland
#>  date     2022-09-20
#>  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package      * version    date (UTC) lib source
#>  assertthat     0.2.1      2019-03-21 [1] CRAN (R 4.2.0)
#>  cli            3.3.0.9000 2022-06-15 [1] Github (r-lib/cli@31a5db5)
#>  codetools      0.2-18     2020-11-04 [1] CRAN (R 4.2.0)
#>  colorspace     2.0-3      2022-02-21 [1] CRAN (R 4.2.0)
#>  conmat       * 0.0.0.9004 2022-09-20 [1] local
#>  countrycode    1.4.0      2022-05-04 [1] CRAN (R 4.2.0)
#>  curl           4.3.2      2021-06-23 [1] CRAN (R 4.2.0)
#>  data.table     1.14.2     2021-09-27 [1] CRAN (R 4.2.0)
#>  DBI            1.1.3      2022-06-18 [1] CRAN (R 4.2.0)
#>  digest         0.6.29     2021-12-01 [1] CRAN (R 4.2.0)
#>  dplyr          1.0.9      2022-04-28 [1] CRAN (R 4.2.0)
#>  ellipsis       0.3.2      2021-04-29 [1] CRAN (R 4.2.0)
#>  evaluate       0.16       2022-08-09 [1] CRAN (R 4.2.0)
#>  fansi          1.0.3      2022-03-24 [1] CRAN (R 4.2.0)
#>  farver         2.1.1      2022-07-06 [1] CRAN (R 4.2.0)
#>  fastmap        1.1.0      2021-01-25 [1] CRAN (R 4.2.0)
#>  fs             1.5.2      2021-12-08 [1] CRAN (R 4.2.0)
#>  furrr          0.3.1      2022-08-15 [1] CRAN (R 4.2.0)
#>  future         1.27.0     2022-07-22 [1] CRAN (R 4.2.0)
#>  generics       0.1.3      2022-07-05 [1] CRAN (R 4.2.0)
#>  ggplot2        3.3.6      2022-05-03 [1] CRAN (R 4.2.0)
#>  globals        0.16.0     2022-08-05 [1] CRAN (R 4.2.0)
#>  glue           1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
#>  gtable         0.3.0      2019-03-25 [1] CRAN (R 4.2.0)
#>  highr          0.9        2021-04-16 [1] CRAN (R 4.2.0)
#>  hms            1.1.1      2021-09-26 [1] CRAN (R 4.2.0)
#>  htmltools      0.5.3      2022-07-18 [1] CRAN (R 4.2.0)
#>  httr           1.4.4      2022-08-17 [1] CRAN (R 4.2.0)
#>  jsonlite       1.8.0      2022-02-22 [1] CRAN (R 4.2.0)
#>  knitr          1.40       2022-08-24 [1] CRAN (R 4.2.0)
#>  labeling       0.4.2      2020-10-20 [1] CRAN (R 4.2.0)
#>  lattice        0.20-45    2021-09-22 [1] CRAN (R 4.2.0)
#>  lifecycle      1.0.2      2022-09-09 [1] CRAN (R 4.2.0)
#>  listenv        0.8.0      2019-12-05 [1] CRAN (R 4.2.0)
#>  lubridate      1.8.0      2021-10-07 [1] CRAN (R 4.2.0)
#>  magrittr       2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
#>  Matrix         1.4-1      2022-03-23 [1] CRAN (R 4.2.0)
#>  mgcv           1.8-40     2022-03-29 [1] CRAN (R 4.2.0)
#>  mime           0.12       2021-09-28 [1] CRAN (R 4.2.0)
#>  munsell        0.5.0      2018-06-12 [1] CRAN (R 4.2.0)
#>  nlme           3.1-159    2022-08-09 [1] CRAN (R 4.2.0)
#>  oai            0.3.2      2021-05-13 [1] CRAN (R 4.2.0)
#>  parallelly     1.32.1     2022-07-21 [1] CRAN (R 4.2.0)
#>  patchwork      1.1.1      2020-12-17 [1] CRAN (R 4.2.0)
#>  pillar         1.8.1      2022-08-19 [1] CRAN (R 4.2.0)
#>  pkgconfig      2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
#>  plyr           1.8.7      2022-03-24 [1] CRAN (R 4.2.0)
#>  purrr        * 0.3.4      2020-04-17 [1] CRAN (R 4.2.0)
#>  R.cache        0.16.0     2022-07-21 [1] CRAN (R 4.2.0)
#>  R.methodsS3    1.8.2      2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo           1.25.0     2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils        2.12.0     2022-06-28 [1] CRAN (R 4.2.0)
#>  R6             2.5.1      2021-08-19 [1] CRAN (R 4.2.0)
#>  RColorBrewer   1.1-3      2022-04-03 [1] CRAN (R 4.2.0)
#>  Rcpp           1.0.9      2022-07-08 [1] CRAN (R 4.2.0)
#>  readr          2.1.2      2022-01-30 [1] CRAN (R 4.2.0)
#>  reprex         2.0.1      2021-08-05 [1] CRAN (R 4.2.0)
#>  rlang          1.0.5      2022-08-31 [1] CRAN (R 4.2.0)
#>  rmarkdown      2.16       2022-08-24 [1] CRAN (R 4.2.0)
#>  rstudioapi     0.13       2020-11-12 [1] CRAN (R 4.2.0)
#>  scales         1.2.0      2022-04-13 [1] CRAN (R 4.2.0)
#>  sessioninfo    1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
#>  socialmixr     0.1.8      2020-11-14 [1] CRAN (R 4.2.0)
#>  stringi        1.7.8      2022-07-11 [1] CRAN (R 4.2.0)
#>  stringr        1.4.1      2022-08-20 [1] CRAN (R 4.2.0)
#>  styler         1.7.0      2022-03-13 [1] CRAN (R 4.2.0)
#>  tibble         3.1.8      2022-07-22 [1] CRAN (R 4.2.0)
#>  tidyr          1.2.0      2022-02-01 [1] CRAN (R 4.2.0)
#>  tidyselect     1.1.2      2022-02-21 [1] CRAN (R 4.2.0)
#>  tzdb           0.3.0      2022-03-28 [1] CRAN (R 4.2.0)
#>  utf8           1.2.2      2021-07-24 [1] CRAN (R 4.2.0)
#>  vctrs          0.4.1      2022-04-13 [1] CRAN (R 4.2.0)
#>  withr          2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
#>  wpp2015        1.1-2      2019-01-07 [1] CRAN (R 4.2.0)
#>  xfun           0.33       2022-09-12 [1] CRAN (R 4.2.0)
#>  XML            3.99-0.10  2022-06-09 [1] CRAN (R 4.2.0)
#>  xml2           1.3.3      2021-11-30 [1] CRAN (R 4.2.0)
#>  yaml           2.3.5      2022-02-21 [1] CRAN (R 4.2.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Review by njtierney — CHANGES_REQUESTED** [review]
So the next step is to now replace all instances of `plot_matrix` and `plot_setting_matrices` with `autoplot` - so examples in the vignettes, and the README and anywhere else

**Review by njtierney — CHANGES_REQUESTED** [review]
Nice work! A couple of small things:

- For `plot_matrix` and `plot_setting_matrices`:
  -  Remove `@export`
  - Add `@noRd` and `@note internal`
  - Cut and paste the help file notes into autoplot helpfile - this might involve combining some of the text from `plot_matrix` and `plot_setting_matrices`

- Can you add vdiffr tests to check that the `autoplot` functions work? See https://github.com/ropensci/visdat/blob/master/tests/testthat/test-vis-dat.R for an example of how to do this - you should be able to use some of the code from the examples to create objects to visualise - if possible try and reduce computation time as much as possible.

File: tests/testthat/_snaps/get-polymod-population.md [hunk 1]
```diff
@@ -27,18 +26,17 @@
...
+       2               5   1968449.
+       3              10   2138897.
+       4              15   2312032.
+       5              20   2407486.
+       6              25   2423602.
+       7              30   2585137.
+       8              35   2969393.
+       9              40   3041663.
+      10              45   2809154.
       # ... with 11 more rows
```
Comment (njtierney) [inline tests/testthat/_snaps/get-polymod-population.md:39]: I think we discussed this in our last meeting - are these numbers different due to some random seed issue or something?

**Review by njtierney — COMMENTED** [review]
I think this all looks great now! Just one minor comment regarding the differing numbers in `get_polymod_population()`

**Comment by njtierney** [comment]
This looks good to merge to me - the CI tests failing appear to be some strange GH actions issue unrelated to the package - as it passed on Mac and ubuntu fine

**Pull request merged**